docs/arch/adr-003-versioning-branching-strategy.md

```diff
@@ -0,0 +1,108 @@
+# Release Versioning / Branching Strategy
+
+## Context
+With `dbt` ever evolving and progressing, we continue to ship and release versions on a regular basis. As we release `dbt`, versioning and branching become key to identifying what is different for users and conveying the expectations of those changes. We currently push all our commits to `main` which is the latest version of our code. We don't always want to release the latest version of our code though. We want to be able to isolate commits to go into particular releases based on urgency (e.g. bug fixes) and stability (e.g. new features). 
+
+We already have an established versioning strategy when it comes to major/minor/patch releases. The following is our approach as to what to expect from each release version:
+
+* Major version: This signifies that breaking changes which require user action are included in this release.
+* Minor version: This signifies that new features are included in this release.
+* Patch version: This signifies that bug and security fixes are included in this release.
+
+We need a versioning strategy that allows us to to convey to users the confidence and stability of code they are installing. We want a versioning strategy that allows users to try out new changes, and experiment with them to get early feedback. We want to make sure users can also get a preview of new features we are working on, so they can incorporate these changes into their work as soon as possible. 
+
+We need a branching strategy to support the goals of the versioning strategy. We do not want to have to stop development on `main` as we await a release, nor do we want every change in `main` published in every release. Using branches, we can decide which changes go into which release.
+
+## Common Terminology
+* Release version: refers to the numeric value of the release (eg. `1.0.1`, `0.21.0`) There are major, minor, and patch release versions that map to standard [semantic versioning practices](https://semver.org/#semantic-versioning-200).
+* Release phase: refers to the stability of the release (eg. beta, rc, final). 
+
+## Requirements
+The following are requirements that must be met for releasing dbt:
+
+1. We must have major, minor, and patch releases.
+1. We must have release phases that have the goal of garnering early feedback from the Community.
+1. We must have release phases that aim to identify bugs and test for stability.
+1. We must have release phases that are stable and of production quality.
+1. We must have the ability to isolate changes from different versions and phases to release.
+
+## Branching Strategy Options
+The 2 branching strategies we are considering using going forward are:
```

Comment:
I think this should be more like "alternatives we considered" since we're continuing with our current strategy after talking about it.

Reply:
What about "Branching Strategies Considered"? In ADR's I've seen, it's more about listing out the options in the first half of the doc and then the decisions made and why in the second half. I like having the 2 strategies juxtaposed here to see the pros and cons. I do call out the current strategy so we remember going forward that we kept with the existing position but this time we knows the reason why it was chosen